addons/mail/models/discuss/res_partner.py

```diff
@@ -90,5 +102,15 @@ def get_mention_suggestions_from_channel(self, channel_id, search, limit=8):
                 "persona": {"partner": {"id": True}},
             },
         )
-        store.add(partners)
+        store.add("Thread", {
+            "id": channel.id,
+            "model": "discuss.channel",
+            "authorizedGroupId": channel.group_public_id.id or None
+        })
+        for p in partners:
+            store.add("Persona",{
+                "id": p.id,
+                'type': "partner",
+                "group_id": next((group.id for group in p.user_ids.groups_id if group.id == channel.group_public_id.id), None)
```

Comment:
Syntax to add in store was improved in 18.0 (can pass record immediately rather than named model name). I would retarget the PR to 18.0 for simpler diff.

The PR is stated as a fix, which is ok to make improvement in 18.0. However the limitation was always there and it wasn't really a bug more a design choice with simpler discuss code.

17.4 is short-lived so better focus work on 18.0 with newer syntax and has to live for 3 years.